### DISCUSSION TITLE
Can't get AdGuard-home to work for docker containers

### DISCUSSION THREAD

**Discussion by JCPvanHooren** [AdguardTeam/AdGuardHome#6252] [Q&A]
Today I had to remove and re-install my AdGuard-home container.

I've removed the container, pruned docker and reinstalled AdGuard-home in a docker container.
I expected it to work as it used to, but unfortunately now my other docker containers fail to resolve dns requests.
I do see the requests being processed by AdGuard, but the application in another container seems to time out on dns.

I have my router give all clients the IP-address of my NAS as DNS server. This also means my NAS get's its own IP-address as DNS server and that seems to work if I ping a url from SSH and across the devices on my network. When I ping an address from within a docker container however, it won't resolve (`ping: bad address` error). Using `nslookup` it times out (`';; communications error to 192.168.1.104#53: timed out`, where 192.168.1.104 is my NAS's IP, so should be the correct IP for the DNS server)

The only way to fix it seems to manually overwrite the DHCP DNS in my Synology to use the router as DNS instead of AdGuard. But that defeats the purpose of using AdGuard as DNS.

Does anyone have any thoughts on how to resolve?
I've tried so many things now, that I can't list them, but happy to engage in some trouble shooting.

**Comment by antoniozh** [comment]
Hi, you can bind your adguard 53 port to the internet interface, it seems that some routing issues cause adguard to respond via the wrong interface when all interfaces are bound. There's a similar issue I experienced in PiHole that worked the same. 

https://dockerlabs.collabnix.com/intermediate/networking/Configuring_DNS.html
https://discourse.pi-hole.net/t/solve-dns-resolution-in-other-containers-when-using-docker-pihole/31413/2

In my example, my compose ports section looks like this: 
```
    ports: 
      - "192.168.1.2:53:53/tcp"
      - "192.168.1.2:53:53/udp"
      - "[::]:53:53"
      - 443:443/udp
      - 3000:3000/tcp 

```

**Comment by JCPvanHooren** [comment]
Thank you @antoniozh for reminding me this issue was still open.
I don't know how I solved it, but I did.
Could be, as you said, by binding to port 53 as my current config looks similar.
